Userland/Libraries/LibWeb/Page/EventHandler.cpp

```diff
@@ -545,7 +545,41 @@ bool EventHandler::handle_doubleclick(Gfx::IntPoint const& position, unsigned bu
     auto offset = compute_mouse_event_offset(position, *layout_node);
     node->dispatch_event(UIEvents::MouseEvent::create_from_platform_event(UIEvents::EventNames::dblclick, offset.x(), offset.y(), position.x(), position.y(), button));
 
-    // FIXME: Select word
+    // NOTE: Dispatching an event may have disturbed the world.
+    if (!paint_root() || paint_root() != node->document().paint_box())
+        return true;
+
+    if (button == GUI::MouseButton::Primary) {
+        if (auto result = paint_root()->hit_test(position.to_type<float>(), Painting::HitTestType::TextCursor); result.has_value()) {
+            auto paintable = result->paintable;
+            if (!paintable->dom_node())
+                return true;
+
+            auto const& layout_node = paintable->layout_node();
+            if (!layout_node.is_text_node())
+                return true;
+            auto const& text_for_rendering = verify_cast<Layout::TextNode>(layout_node).text_for_rendering();
+
+            int first_word_break_before = [&] {
+                for (size_t i = result->index_in_node - 1; i > 0; --i) {
+                    if (text_for_rendering[i] == ' ')
```

Comment:
This should probably check if the character is whitespace instead. For example, double-clicking on "mousemove" in the mouse-events test page highlighted this, because it doesn't treat the newline as whitespace:
![image](https://user-images.githubusercontent.com/222642/174083303-5f8ddbcf-7155-4999-9df8-66e04f754bf4.png)